elrond-wasm-debug/src/ext_mock.rs

```diff
@@ -453,6 +482,19 @@ impl elrond_wasm::ContractIOApi<RustBigInt, RustBigUint> for TxContext {
         }
     }
 
+    fn get_argument_u64(&self, arg_index: i32) -> u64 {
+        let bytes = self.get_argument_vec_u8(arg_index);
+        let bu = BigUint::from_bytes_be(&bytes);
+        if let Some(v) = bu.to_u64() {
+            v
+        } else {
+            panic!(TxPanic{
```

Comment:
this panic simulates "signal_error" in the mock, which terminates the process early.
It's the only way I found to simulate that behaviour.
The mock then wraps the execution into a panic unwind closure, so it can catch the panic. Rust doesn't really have try-catch, but you can force something like that with panic unwind.

Reply:
This particular error would be thrown directly in arwen, so the contract wouldn't see it. That's why there's no classical Result::Err being returned

Reply:
I did think about having a length check before letting arwen crash it, so we can get a better error message, but that makes the contract bigger, so there's that. Can be changed if we feel like